src/main/java/io/druid/data/input/impl/JSONParseSpec.java

```diff
@@ -49,18 +59,53 @@ public void verify(List<String> usedCols)
   @Override
   public Parser<String, Object> makeParser()
   {
-    return new JSONParser(objectMapper, null);
+    if(flattenSpec != null) {
```

Comment:
If possible I think we should always use the JSONPathParser (so, if there is no flattenSpec, use an empty fieldSpecs list and always enable discovery). I think for it to be possible we need:

- `new JSONPathParser(ImmutableList.of(), true, objectMapper)` should behave the same as `new JSONParser(objectMapper, null)`
- performance should be similar or better